test/curlx_test.cpp

```diff
@@ -5,7 +5,14 @@
 #include "libndt7/internal/curlx.hpp"
 
 #define CATCH_CONFIG_MAIN
+#ifndef __clang__
```

Comment:
```suggestion
// We need to disable -Wmaybe-uninitialized because of https://github.com/m-lab/ndt7-client-cc/pull/9. We
// may want to check whether we could remove this restriction in the future.
#ifndef __clang__
```

I suggest adding this comment here and in the other files, so we remember why we added these diagnostic.